### PR TITLE
fix crash omnibox search banner android

### PR DESCRIPTION
**Pull request by tapanmodh**
Resolves https://github.com/brave/brave-browser/issues/25059

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:
Please follow the issue ticket.

### CODE REVIEW

**Review by SergeyZhukovsky — APPROVED** [review]
++

File: android/java/apk_for_test.flags [hunk 1]
```diff
@@ -507,8 +507,6 @@
 }
 
 -keep class org.chromium.chrome.browser.omnibox.suggestions.DropdownItemViewInfoListBuilder {
-    *** mDropdownHeight;
```
Comment (samartnik) [inline android/java/apk_for_test.flags:510]: I suppose changes in `brave/build/android/bytecode/java/org/brave/bytecode/BraveDropdownItemViewInfoListBuilderClassAdapter.java` are not necessary as well anymore?

**Review by samartnik — CHANGES_REQUESTED** [review]

File: android/java/apk_for_test.flags (hunk 1 above)
Comment (tapanmodh) [reply android/java/apk_for_test.flags:510]: Yes, thanks for pointing out. removing now

**Review by deeppandya — APPROVED** [review]
LGTM. Verified the change with the build from the branch.

**Comment by kjozwiak** [comment]
Verification **PASSED** on `Samsung S10+` running `Android 12` using the following build(s):

```
Brave | 1.45.12 Chromium: 105.0.5195.68 (Official Build) canary (32-bit)
--- | ---
Revision | ad13e82529051bac6a0e65f455e6d7a1e5fd7938-refs/branch-heads/5195@{#903}
OS | Android 12; Build/SP1A.210812.016
```

Basically went through the STR/Cases outlined via https://github.com/brave/brave-core/pull/14350#issuecomment-1231063327 and ensured that the feature was working. Also verified that https://github.com/brave/brave-browser/issues/25059#issue-1355086510 was also addressed/fixed.

### `Test Case #1` (Supported Regions - US/CA/DE/FR/UK/AT/ES/MX)

* installed `1.45.12 Chromium: 105.0.5195.68`
* ensured that the system language is set to either `US`, `CA`, `DE`, `FR`, `UK`, `AT`, `ES` or `MX`
* launched brave and changed the default SE from `Brave` to anything else within the list (Google, DDG, Startpage etc..)
* added `--enable-features=BraveSearchOmniboxBanner` into `Command Line String` via `QA Preferences`
* re-launched the browser tapped on the URL bar and ensured that the Brave search promo is displayed via the omnibox
* ensured that the Brave search promo appears when tapping/editing a URL and the promo hasn't been closed/dismissed
* ensured that the Brave search promo doesn't appear if `Private` is set to another SE but `Standard` is still selected as `Brave`
* ensured that tapping on `Maybe Later` closes the promo within the omnibox and doesn't appear again within the session
  * ensured that it doesn't appear via the omnibox via NTP when starting to enter a URL/search term
  * ensured that tapping on a URL in an existing tab doesn't display the promo
* ensured that tapping on `Try Brave Search` opens `https://search.brave.com/search?q=motorsport&action=makeDefault`
  * ensured that the search term is also appended as per `search?q=motorsport`
  * ensured that tapping on `Try Brave Search` adds `action=makeDefault` into the URL

**`Austria`**

**`Example`** | **`Example`** | **`Example`** |  **`Example`** | **`Example`** | **`Example`** | **`Example`** | **`Example`** | **`Example`** 
---------------|-----------------|----------------|----------------|-----------------|----------------|---------------|-------|--------
Screenshot_20220830-164331_Brave - Nightly | Screenshot_20220830-164342_Brave - Nightly | Screenshot_20220830-164413_Brave - Nightly | Screenshot_20220830-164514_Brave - Nightly | Screenshot_20220830-164543_Brave - Nightly | Screenshot_20220830-164706_Brave - Nightly | Screenshot_20220830-164639_Brave - Nightly | Screenshot_20220830-164722_Brave - Nightly | Screenshot_20220830-164727_Brave - Nightly

**`Spain`**

**`Example`** | **`Example`** | **`Example`** |  **`Example`** | **`Example`** | **`Example`** | **`Example`** | **`Example`** | **`Example`** 
---------------|-----------------|----------------|----------------|-----------------|----------------|---------------|-------|--------
Screenshot_20220830-213504_Brave - Nightly | Screenshot_20220830-214816_Brave - Nightly | Screenshot_20220830-213612_Brave - Nightly | Screenshot_20220830-213627_Brave - Nightly | Screenshot_20220830-213644_Brave - Nightly | Screenshot_20220830-213707_Brave - Nightly | Screenshot_20220830-213716_Brave - Nightly | Screenshot_20220830-213751_Brave - Nightly | Screenshot_20220830-213758_Brave - Nightly

### `Test Case #2` (Regions that don't have Brave Search as default SE/Still use SE onboarding)

* installed `1.45.12 Chromium: 105.0.5195.68`
* ensured that the system language is set to the correct locale (in this case, either `Italian` or `Chinese`)
* launched brave and added `--enable-features=BraveSearchOmniboxBanner` into `Command Line String` via `QA Preferences`
* re-launched the browser tapped on the URL bar twice and ensured that the SE onboarding screen is displayed
* ensured that Google was selected as the default via the onboarding screen
* ensured that Google is default SE for both `Standard` & `Private` windows once the onboarding screen has been dismissed
* ensured that the Brave search promo isn't appearing via the omnibox when a text/search term is entered

**`Italy`**

**`Example`** | **`Example`** | **`Example`** |  **`Example`**
---------------|-----------------|----------------|----------------
Screenshot_20220830-215454_Brave - Nightly | Screenshot_20220830-215505_Brave - Nightly | Screenshot_20220830-215515_Brave - Nightly | Screenshot_20220830-215528_Brave - Nightly

**`China`**

**`Example`** | **`Example`** | **`Example`** |  **`Example`**
---------------|-----------------|----------------|----------------
Screenshot_20220830-220149_Brave - Nightly | Screenshot_20220830-220202_Brave - Nightly | Screenshot_20220830-220212_Brave - Nightly | Screenshot_20220830-220219_Brave - Nightly

### `Test Case #3` (Maybe Later & Dismiss)

* ensured that closing the Brave search promo via `Maybe Later` doesn't re-appear in the same session
  * ensure the promo is not re-appearing when typing into the omnibox via NTP
  * ensure the promo  is not re-appearing when taping on a URL on an existing opened tab
* ensured that the promo re-appears again when the browser is re-launched after tapping on `Maybe Later`
  * ensure that the `Dismiss` button is visible and not the `Maybe Later` as per https://github.com/brave/brave-core/pull/14350#issue-1319129566
* ensured that once a user taps/clicks on `Dismiss`, the promo never re-appears (tried via several re-launches)

**`Example`** | **`Example`** | **`Example`** |  **`Example`**
---------------|-----------------|----------------|----------------
Screenshot_20220830-220711_Brave - Nightly | Screenshot_20220830-220716_Brave - Nightly | Screenshot_20220830-220727_Brave - Nightly | Screenshot_20220830-220731_Brave - Nightly

### `Test Case #4` (Automatically dismiss after 15 days)

* installed `1.45.12 Chromium: 105.0.5195.68`
* ensured that the system language is set to either `US`, `CA`, `DE`, `FR`, `UK`, `AT`, `ES` or `MX`
* launched brave and changed the default SE from `Brave` to anything else within the list (Google, DDG, Startpage etc..)
* added `--enable-features=BraveSearchOmniboxBanner` into `Command Line String` via `QA Preferences`
* re-launched the browser tapped on the URL bar and ensured that the Brave search promo is displayed via the omnibox
* moved the time/date on the device ahead ~16 days and re-launched the browser
* ensured that the Brave search promo doesn't appear via the omnibox when tapping via NTP
* ensured that the Brave search promo doesn't appear when tapping on the omnibox if an already opened tab

**Comment by kjozwiak** [comment]
Also found https://github.com/brave/brave-browser/issues/25091 while running through the above which should also get fixed for `1.43.x`.

**Pull request merged**